crates/evm/src/execute.rs

```diff
@@ -60,56 +63,91 @@ pub struct EthBlockOutput<T> {
     pub gas_used: u64,
 }
 
+/// The output of a batch of ethereum blocks.
+#[derive(Debug)]
+pub struct BatchBlockExecutionOutput {
+    /// Bundle state with reverts.
+    pub bundle: BundleState,
+    /// The collection of receipts.
+    /// Outer vector stores receipts for each block sequentially.
+    /// The inner vector stores receipts ordered by transaction number.
+    ///
+    /// If receipt is None it means it is pruned.
+    pub receipts: Receipts,
+    /// First block of bundle state.
+    pub first_block: BlockNumber,
+}
+
 /// A helper type for ethereum block inputs that consists of a block and the total difficulty.
 #[derive(Debug)]
-pub struct EthBlockExecutionInput<'a, Block> {
+pub struct BlockExecutionInput<'a, Block> {
```

Comment:
Ah I guess this will be moved over as a type to the `ethereum` crate and be wired to the types. I really don't like the total_difficulty param as it's confusing people but we seem to have a path to getting rid of ti